server/api/src/rsDataObjPut.cpp

```diff
@@ -123,6 +115,9 @@ namespace
         reg_param[REPL_STATUS_KW] = std::to_string(STALE_REPLICA);
         reg_param[STALE_ALL_INTERMEDIATE_REPLICAS_KW] = "";
         if (cond_input.contains(ADMIN_KW)) {
+            // TODO Shouldn't this always be set?
+            // The user's perms would need to be elevated as well.
+            // rsModDataObjMeta should never fail due to perms here.
             reg_param[ADMIN_KW] = "";
         }
```

Comment:
Shouldn't `ADMIN_KW` always be set here? The existence check seems unnecessary.

Reply:
it's just propagating it if it exists, right now, right?   so the question is what bad can happen if we just always set it?

Reply:
this code block's job is to update the db with the correct info. imo, the update should never fail because of the user's lack of access level.

i guess the real question is whether a non-admin user can successfully execute this code path without error.

Reply:
Could it fail due to permissions because the permissions model for opening is different from that of modifying the system metadata (i.e. `rsModDataObjMeta`)? Certainly not a normal (or even fully defensible) use case, but... possible?

Reply:
I hit this error case before. Looking back now, I can see it was because the `ADMIN_KW` was set while I was a `rodsuser`. 
https://github.com/irods/irods/blob/e4fd2a6a59dbc85c376e30e0481d8f3339da2893/plugins/database/src/db_plugin.cpp#L2206-L2211

I think this is okay.